website/pages/docs/secrets/mongodbatlas/index.mdx

```diff
@@ -0,0 +1,193 @@
+$ vault write mongodbatlas/roles/test \
+    project_id=5cf5a45a9ccf6400e60981b6 \
+    roles=GROUP_DATA_ACCESS_READ_ONLY
+```
+
+## Programmatic API Key Whitelist
+
+Programmatic API Key access can and should be limited with a IP Whitelist. In the following example both a CIDR
+block and IP address are added to the IP whitelist for Keys generated with this Vault role:
+
+```bash
+  $ vault write atlas/roles/test \
+      project_id=5cf5a45a9ccf6400e60981b6 \
+      roles=GROUP_CLUSTER_MANAGER \
+      cidr_blocks=192.168.1.3/32 \
+      ip_addresses=192.168.1.3
+```
+
+Verify the created Programmatic API Key Vault role has the added CIDR block and IP address by running:
+
+  ```bash
+    $ vault read atlas/roles/test
+
+      Key                       Value
+      ---                       -----
+      cidr_blocks               [192.168.1.3/32]
+      ip_addresses              [192.168.1.3]
+      max_ttl                   0s
+      organization_id           n/a
+      roles                     [GROUP_CLUSTER_MANAGER]
+      project_id                5cf5a45a9ccf6400e60981b6
+      roles                     n/a
+      ttl                       0s
+  ```
+
+## TTL and Max TTL
+
+Programmatic API Keys Vault have a time-to-live (TTL) and maximum time-to-live (Max TTL).
+When a credential expires it's automatically revoked. You can set the TTL and Max TTL for each role
+or by tuning the secrets engine's configuration.
+
+The following creates a Vault role "test" for a Project level Programmatic API key with a 2 hour time-to-live and a
+max time-to-live of 5 hours.
+
+```bash
+$ vault write mongodbatlas/roles/test \
+    project_id=5cf5a45a9ccf6400e60981b6 \
+    roles=GROUP_DATA_ACCESS_READ_ONLY \
+    ttl=2h \
+    max_ttl=5h
+```
+
+You can verify the role that you have created with:
+
+```bash
+$ vault read mongodbatlas/roles/test
+
+    Key                       Value
+    ---                       -----
+    organization_id           5b71ff2f96e82120d0aaec14
+    roles                     [GROUP_DATA_ACCESS_READ_ONLY]
+    project_id                5cf5a45a9ccf6400e60981b6
+    roles                     n/a
+    ttl                       2h0m0s
+    max_ttl                   5h0m0s
+```
+
+
+## Generating credentials
+
+After a user has authenticated to Vault has has sufficient permissions, a read request to the
+`creds` endpoint for the role will generate and return new Programmatic API Keys:
+
+```bash
+$ vault read mongodbatlas/creds/test
+
+    Key                Value
+    ---                -----
+    lease_id           mongodbatlas/creds/test/0fLBv1c2YDzPlJB1PwsRRKHR
+    lease_duration     2h
+    lease_renewable    true
+    description        vault-test-1563980947-1318
+    private_key        905ae89e-6ee8-40rd-ab12-613t8e3fe836
+    public_key         klpruxce
+```
+
+## API
+
+The MongoDB Atlas secrets engine has a full HTTP API. Please see the
+[MongoDB Atlas secrets engine API docs](/api/secret/mongodbatlas) for more details.
```

Comment:
This link is broken. I'm not sure where it's intending to go, maybe here?
```
[MongoDB Atlas Database Plugin HTTP API](/api-docs/secret/databases/mongodbatlas)
```

Reply:
The link is correct, but we don't have content yet.